sdk/monitor/azure-monitor-query/src/main/java/com/azure/monitor/query/implementation/metrics/models/MetricsHelper.java

```diff
@@ -268,4 +316,60 @@ public static String getSubscriptionFromResourceId(String s) {
         return subscriptionId;
     }
 
+    /**
+     * Converts a {@link QueryTimeInterval} to ISO 8601 string format suitable for Azure Monitor Metrics API.
+     * For duration-only intervals, this method converts them to absolute start/end times based on current time.
+     *
+     * @param timeInterval The time interval to convert.
+     * @return ISO 8601 formatted string representation with absolute start/end times.
+     */
+    public static String toMetricsTimespan(QueryTimeInterval timeInterval) {
+        if (timeInterval == null) {
+            return null;
+        }
+
+        // If we have both start and end times, use them directly
+        if (timeInterval.getStartTime() != null && timeInterval.getEndTime() != null) {
+            return formatForMetrics(timeInterval.getStartTime()) + "/" + formatForMetrics(timeInterval.getEndTime());
+        }
+
+        // If we have start time and duration, calculate end time
+        if (timeInterval.getStartTime() != null && timeInterval.getDuration() != null) {
+            return formatForMetrics(timeInterval.getStartTime()) + "/"
+                + formatForMetrics(timeInterval.getStartTime().plus(timeInterval.getDuration()));
+        }
+
+        // If we have duration and end time, calculate start time
+        if (timeInterval.getDuration() != null && timeInterval.getEndTime() != null) {
+            return formatForMetrics(timeInterval.getEndTime().minus(timeInterval.getDuration())) + "/"
+                + formatForMetrics(timeInterval.getEndTime());
+        }
+
+        // If we only have duration, calculate absolute start and end times based on current time
+        if (timeInterval.getDuration() != null) {
+            OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
+            OffsetDateTime startTime = now.minus(timeInterval.getDuration());
+            return formatForMetrics(startTime) + "/" + formatForMetrics(now);
+        }
+
+        return null;
+    }
+
+    /**
+     * Formats an OffsetDateTime for Azure Monitor Metrics API.
+     * Uses ISO format but omits seconds when they are zero.
+     *
+     * @param dateTime The datetime to format.
+     * @return Formatted string.
+     */
+    private static String formatForMetrics(OffsetDateTime dateTime) {
+        if (dateTime.getSecond() == 0 && dateTime.getNano() == 0) {
+            // Format without seconds when they are zero: 2025-01-01T00:00Z
+            return dateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mmXXX"));
+        } else {
```

Comment:
Why do we need to special case this? Can we just do `dateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME)` always?